beancount_interpolate/recur.py

```diff
@@ -1,34 +1,33 @@
 __author__ = 'Akuukis <akuukis@kalvis.lv>'
 import datetime
-import math
 
-from beancount.core.amount import Amount
 from beancount.core.data import filter_txns
 from beancount.core.number import D
 
 from .common import extract_mark_tx
 from .common import new_whole_entries
 from .common import read_config
 from .common import parse_mark
+from .common import get_number_of_txn
 
 __plugins__ = ['recur']
 
 
 def dublicate_over_period(params, default_date, value, config):
     begin_date, duration, step = parse_mark(params, default_date, config)
-    period = math.floor( duration / step )
+    period = get_number_of_txn(begin_date, duration, step, config['max_new_tx'])
 
     if(period > config['max_new_tx']):
         period = config['max_new_tx']
         duration = period * step
```

Comment:
this block seems redundant now, as you're handling `max_new_tx` within `get_number_of_txn`.